reference/mixin-arguments.rst

```diff
@@ -27,3 +27,22 @@ The following arguments are provided by the ``colcon-mixin`` package:
   * **debug**:
 
     - ``cmake-args``: ``['-DCMAKE_BUILD_TYPE=Debug']``
+
+It is important to note that in their current implementation, mixins do not
+possess semantic knowledge about CMake arguments that would allow them to
+execute merges of **arguments that appear multiple times** correctly.  Those
+conditions arise, for example, when **mixin1** and **mixin2** contain
+overlapping parameters, or when any **mixin** specifies an argument that also
+appears on the command line.  This situation should be avoided since one of
+the parameters will be silently suppressed and the outcome not be as expected.
```

Comment:
Based on the comments below I think the description of the limitation needs to be more precise / specific, e.g. what is meant with "overlapping parameters".

The problematic case mentioned in the referenced ticket is if both mixins (or a mixin and the CLI) contain the same CMake argument (e.g. `-DCMAKE_C_FLAGS=..`) but with different values. Only in that case `colcon` doesn't know about the semantic of `-DCMAKE_C_FLAGS` and that it might make sense the to values of the CMake definition.